opsmngr/live_migration.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const liveMigrationBasePath = "api/public/v1.0/orgs/%s/liveExport/migrationLink"
+
+// LiveMigrationService is an interface for interfacing with the Live Migration
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/cloud-migration/
+type LiveMigrationService interface {
+	Create(context.Context, string, *atlas.LinkToken) (*atlas.LiveMigration, *Response, error)
+	Delete(context.Context, string) (*Response, error)
+	Get(context.Context, string) (*atlas.LiveMigration, *Response, error)
+}
+
+// LiveMigrationOp provides an implementation of the LiveMigrationService interface.
+type LiveMigrationServiceOp service
+
+var _ LiveMigrationService = &LiveMigrationServiceOp{}
+
+// Get returns the status of the connection between the specified source Ops Manager organization and the target MongoDB Atlas organization.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/cloud-migration/return-the-status-of-the-organization-link/
+func (s *LiveMigrationServiceOp) Get(ctx context.Context, orgID string) (*atlas.LiveMigration, *Response, error) {
```

Comment:
I don't fear what @colm-quinn mentions on the clients as they move together, unlike atlas and ops manager, so their versions are (usually) hand in hand with changes but... I don't think these two to be the same

OM: 
```json
{
  "status": "SYNCED"
}
```

Atlas
```json
{
"_id": "7e68e12770616e75e6df43d0",
"status": "NEW",
"migrationHosts": "vm001.example.com"
}
```

Reply:
also the name `LiveMigration` is misleading since this is only connection status and not the status of a migration